http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -397,6 +422,18 @@ private HttpRequest buildJettyRequest(Request finalRequest)
         jettyRequest.timeout(requestTimeoutMillis, MILLISECONDS);
         jettyRequest.idleTimeout(idleTimeoutMillis, MILLISECONDS);
 
+        // client authentications
+        if (jettyRequest.getURI().getScheme().equalsIgnoreCase("https")) {
+            jettyRequest.header(io.airlift.security.utils.SecurityUtil.REALM_IN_CHALLENGE, "true");
+            AuthenticationStore authStore = httpClient.getAuthenticationStore();
+            if (securityConfig != null && securityConfig.enabled()) {
+                String authScheme = securityConfig.getAuthScheme().toString();
+                if (authStore.findAuthentication(authScheme, jettyRequest.getURI(), null) == null) {
```

Comment:
Why to we need to cache the Authentication objects? `AuthenticationProvider.createAuthentication` just creates a new instance of SpnegoAuthentication. Can't they be created on the fly in `AuthenticationStore.findAuthentication`?